### PR TITLE
Refactor ci github actions

### LINKED ISSUE #2: Refactor CI with GitHub Actions
# Problem Description

- we do not have travis ci installed and we cannot use the original project's CI. Since this project's creation Github Actions matured enough. We should migrate the CI pipeline to Github Actions.
- The new pipeline should take advantage of Make and Docker.
- The pipeline should spit out a Docker image to be used as a base image with other c files ( we need to push the image to Dockerhub or to Github Packages)

# Pipeline stages
- Static Analysis
- Test
- Code Coverage Calculation
- Build
- Deploy

### PR DESCRIPTION
**Pull request by samerbahri98**
This new CI Pipeline should meed our requirements, and now it's safe to deprecate the old one and the old test script

### CODE REVIEW

**Comment by samerbahri98** [comment]
@SABERGLOW @Amani215 please Review/accept the pull request

File: doc/Github Actions.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Refactoring the CI Pipeline
+
+The original CI pipeline only runs the build and the tests. Our CI pipeline runs two jobs
+
+1) It builds the projects, tests it and reports the code coverage
+2) It builds the Docker images and pushes them to Github Packages
+
+These Pipelines run in parallel and isolated from each other (on different build containers).
+
+# Running the CI pipelones
```
Comment (Amani215) [inline doc/Github Actions.md]: pipelines*

File: Makefile [hunk 2]
```diff
@@ -19,15 +21,17 @@ clean:
...
 test:
-	- ( cd test_suite && $(MAKE) all)
-	- ( ./test_suite/test_all )
+	gcc -fprofile-arcs -ftest-coverage test_suite/test_*.c src/libRecommender.a -lm -o test_suite/test_all.o
+	./test_suite/test_all.o
 
 coverage:
-	- ( echo "Coverage is not supported on Alpine Linux" )
+	gcov test_all.c
+	lcov --capture --directory . --output-file coverage.info
```
Comment (Amani215) [inline Makefile:31]: Is it okay if the line that generates html does not exist here?

`genhtml coverage.info --output-directory out`

**Review by Amani215 — CHANGES_REQUESTED** [review]
Check my comments. I left 2 of them.

File: Makefile (hunk 2 above)
Comment (samerbahri98) [reply Makefile:31]: no need

**Pull request merged**